# Analyse Approfondie de ZEvent 2025

## Plan d'Analyse

1.  **Introduction :**
    *   Présentation rapide du ZEvent et du jeu de données.
    *   Objectifs de l'analyse.

2.  **Chargement et Prétraitement des Données :**
    *   Charger tous les jeux de données nettoyés dans des DataFrames pandas.
    *   Effectuer le nettoyage et le prétraitement nécessaires (conversion des types de données, gestion des valeurs manquantes, fusion des jeux de données).

3.  **Analyse Exploratoire des Données (EDA) :**
    *   **Statistiques Globales de l'Événement :**
        *   Total des dons.
        *   Total des spectateurs au fil du temps.
        *   Pic de spectateurs.
        *   Nombre de streamers participants.
    *   **Analyse des Streamers :**
        *   Top streamers par dons, spectateurs moyens et pic de spectateurs.
        *   Distribution des dons et des spectateurs parmi les streamers.
        *   Corrélation entre les spectateurs et les dons.
        *   Analyse de l'efficacité des streamers (dons par heure, dons par follower).
    *   **Analyse Temporelle :**
        *   Évolution des dons et des spectateurs sur la durée de l'événement.
        *   Identifier les moments clés (pics de dons/spectateurs) et les corréler avec le planning de l'événement.
    *   **Analyse des Objectifs de Dons :**
        *   Analyser les objectifs de dons : atteints ou non.
        *   Relation entre les objectifs de dons et la popularité des streamers.

4.  **Analyse Approfondie & Ingénierie de Caractéristiques :**
    *   Créer de nouvelles caractéristiques, telles que `dons_par_spectateur`.
    *   Segmenter les streamers en paliers (par exemple, top-tier, mid-tier, émergents) et comparer leurs performances.
    *   Analyser l'impact des événements du planning officiel du ZEvent sur l'audience et les dons.

5.  **Visualisations :**
    *   Créer une variété de graphiques pour illustrer les résultats :
        *   Graphiques linéaires pour les données temporelles (dons, spectateurs).
        *   Diagrammes à barres pour comparer les streamers.
        *   Nuages de points pour montrer les corrélations.
        *   Histogrammes et boîtes à moustaches for les distributions.
        *   Diagrammes circulaires pour la complétion des objectifs de dons.
        *   Un tableau de bord interactif résumant les métriques clés.

6.  **Conclusion :**
    *   Résumer les principales conclusions de l'analyse.
    *   Fournir des aperçus et des points à retenir des données du ZEvent 2025.
    *   Suggérer des domaines potentiels pour une analyse future.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import os

# Options d'affichage
pd.set_option('display.max_columns', None)
sns.set_theme(style="whitegrid")
plt.rcParams['figure.figsize'] = (12, 6)

## 2. Chargement et Prétraitement des Données

In [2]:
# Définir les chemins vers les fichiers de données
DATA_DIR = '../data/clean'

# Chemins pour les événements
donation_goals_path = os.path.join(DATA_DIR, 'events/donation_goals.csv')
event_schedule_path = os.path.join(DATA_DIR, 'events/event_schedule.csv')

# Chemins pour les streamers
streamers_path = os.path.join(DATA_DIR, 'streamers/zevent_sullygnome_enriched.csv')

# Chemins pour les données temporelles
donations_evolution_path = os.path.join(DATA_DIR, 'temporal/donations_evolution.csv')
viewers_evolution_path = os.path.join(DATA_DIR, 'temporal/viewers_evolution.csv')

# Charger les jeux de données
try:
    donation_goals = pd.read_csv(donation_goals_path)
    event_schedule = pd.read_csv(event_schedule_path)
    streamers = pd.read_csv(streamers_path)
    donations_evolution = pd.read_csv(donations_evolution_path)
    viewers_evolution = pd.read_csv(viewers_evolution_path)
    
    print("Tous les fichiers ont été chargés avec succès.")
except FileNotFoundError as e:
    print(f"Erreur de chargement de fichier : {e}")

# Afficher les premières lignes de chaque dataframe pour vérifier
print("Aperçu des objectifs de dons :")
display(donation_goals.head())
print("\nAperçu du planning de l'événement :")
display(event_schedule.head())
print("\nAperçu des données des streamers :")
display(streamers.head())
print("\nAperçu de l'évolution des dons :")
display(donations_evolution.head())
print("\nAperçu de l'évolution des spectateurs :")
display(viewers_evolution.head())

Tous les fichiers ont été chargés avec succès.
Aperçu des objectifs de dons :


id                      participation_id  \
0  56dbd192-e0ec-437d-94ae-2fd9fa520a61  3f847d41-2642-48a4-8b04-f8fe65fc7c7d   
1  f0744c0d-f88f-4fa0-8e70-e70f0067fceb  3f847d41-2642-48a4-8b04-f8fe65fc7c7d   
2  1838b005-ea33-4900-baac-e1f4c2326fb2  3f847d41-2642-48a4-8b04-f8fe65fc7c7d   
3  56374439-88ec-4327-8df6-1da8b9b8d499  3f847d41-2642-48a4-8b04-f8fe65fc7c7d   
4  c6fc3b52-0652-4177-bb7f-c0123c8f0da4  3f847d41-2642-48a4-8b04-f8fe65fc7c7d   

                                            name category  amount  \
0                            No cut sans top 200   simple   100.0   
1  Deck aléatoire chaque défaite = 20€ (5 games)   simple   250.0   
2                       Participer à la CLASHTAX   simple   500.0   
3                    5 ranked Valo (1 mort = 2€)   simple   750.0   
4                      Photo avant/après (muscu)   simple  1000.0   

   accomplished                                              links  \
0          True  ['https://x.com/Ouahleouff_Just/status/1964112...   
1         False                                                 []   
2         False                                                 []   
3         False                                                 []   
4         False                                                 []   

                            streamer_id streamer_name streamer_login  
0  1ebf92c6-3c34-4062-b3e6-eeae7edc0942    0uahleouff     0uahleouff  
1  1ebf92c6-3c34-4062-b3e6-eeae7edc0942    0uahleouff     0uahleouff  
2  1ebf92c6-3c34-4062-b3e6-eeae7edc0942    0uahleouff     0uahleouff  
3  1ebf92c6-3c34-4062-b3e6-eeae7edc0942    0uahleouff     0uahleouff  
4  1ebf92c6-3c34-4062-b3e6-eeae7edc0942    0uahleouff     0uahleouff


Aperçu du planning de l'événement :


id                            name  \
0  bb5d2165-06c4-4c59-8b01-4f74b2e2e949                         Concert   
1  7a4298da-0a29-47c7-a046-f4e6472e75f5                Lancement ZEVENT   
2  0a04ac88-808b-4fe2-9bd3-8dbade1365ad          Ouverture ZEVENT/Place   
3  f8e6a22a-f349-4ea3-a677-d486f651d56e  Lancement Twitch Plays Pokémon   
4  1b1bd9c6-caad-40ed-ab76-b20b2ebedd57  Mise en vente Cartes collector   

             start_date              end_date  \
0  2025-09-04T18:00:00Z  2025-09-04T21:50:00Z   
1  2025-09-05T16:00:00Z  2025-09-05T16:10:00Z   
2  2025-09-05T16:15:00Z  2025-09-05T16:30:00Z   
3  2025-09-05T16:15:00Z  2025-09-05T16:30:00Z   
4  2025-09-05T17:00:00Z  2025-09-05T17:15:00Z   

                                         description  \
0  Au Zénith de Montpellier avec : IAM, Alonzo, C...   
1                                                NaN   
2  https://place.zevent.fr/ pour placer des pixels !   
3                                                NaN   
4  Rendez-vous sur https://boutique.zevent.fr/ po...   

                                        participants         day  
0  {'host': ['435283ba-af21-41c5-8230-5a8055508ec...  2025-09-04  
1  {'participant': [], 'host': ['dec5b2e1-63e0-47...  2025-09-05  
2                    {'host': [], 'participant': []}  2025-09-05  
3  {'host': ['b5401fdd-990a-4137-8f7f-0ca977b026c...  2025-09-05  
4  {'host': ['435283ba-af21-41c5-8230-5a8055508ec...  2025-09-05


Aperçu des données des streamers :


twitch_login           name   channel_name  total_donations  \
0        domingo        Domingo        Domingo       1479669.67   
1  antoinedaniel  AntoineDaniel  AntoineDaniel       1218838.44   
2        zerator        ZeratoR        ZeratoR       1154211.58   
3     angledroit     AngleDroit     AngleDroit        568758.58   
4          mastu          Mastu          Mastu        553537.29   

           total_viewers_max  duration_seconds  donations_final  \
0  [1757285973691, 259237.0]            2179.0       1478062.67   
1  [1757285973691, 111700.0]            2288.0       1218396.44   
2  [1757285973691, 171999.0]            2319.0       1150918.53   
3   [1757285973691, 17484.0]            2184.0        568653.58   
4   [1757285973691, 77795.0]            1823.0        553328.29   

   donations_start  avg_viewers_zevent  peak_viewers_zevent  watch_time_hours  \
0          7492.90        50170.684685             259237.0        2471463.00   
1          9899.63        30841.936937             111700.0        1459780.00   
2         10386.36        20444.378378             171999.0         923890.25   
3         53940.19         3946.963964              17484.0         175574.00   
4          6985.50        18061.810811              77795.0         832882.75   

   stream_time_hours  peak_viewers_twitch  avg_viewers_twitch  followers  \
0              37.00             253936.0             66796.0  1953343.0   
1              38.50              87263.0             37916.0  1093801.0   
2              39.75             132082.0             23243.0  1628114.0   
3              36.75              14651.0              4778.0   279521.0   
4              31.00              69973.0             26867.0  1350788.0   

   is_partnered  is_mature language  avg_watch_per_stream_hour  global_rank  \
0          True      False   French               66796.297297          1.0   
1          True       True   French               37916.363636          2.0   
2          True      False   French               23242.522013          5.0   
3          True      False   French                4777.523810         20.0   
4          True      False   French               26867.185484          7.0   

   rank_in_page  page  donations_per_hour  donations_per_follower  \
0           1.0   1.0        2.444613e+06                0.757506   
1           2.0   1.0        1.917753e+06                1.114315   
2           5.0   1.0        1.791790e+06                0.708926   
3          20.0   1.0        9.375141e+05                2.034762   
4           7.0   1.0        1.093107e+06                0.409788   

   twitch_efficiency  donations_rank  
0       66796.297297               1  
1       37916.363636               2  
2       23242.522013               3  
3        4777.523810               4  
4       26867.185484               5


Aperçu de l'évolution des dons :


timestamp    donations             datetime
0  1757286000000  16172355.24  2025-09-07 23:00:00
1  1757284200000  15873545.50  2025-09-07 22:30:00
2  1757282400000  15512122.12  2025-09-07 22:00:00
3  1757280600000  13883837.38  2025-09-07 21:30:00
4  1757278800000  13234789.65  2025-09-07 21:00:00


Aperçu de l'évolution des spectateurs :


timestamp   viewers             datetime
0  1757088000000       0.0  2025-09-05 16:00:00
1  1757089800000  348195.0  2025-09-05 16:30:00
2  1757091600000  386638.0  2025-09-05 17:00:00
3  1757093400000  414189.0  2025-09-05 17:30:00
4  1757095200000  417488.0  2025-09-05 18:00:00

### Prétraitement et Nettoyage des Données

In [3]:
# Conversion des colonnes de date en datetime
donations_evolution['datetime'] = pd.to_datetime(donations_evolution['datetime'])
viewers_evolution['datetime'] = pd.to_datetime(viewers_evolution['datetime'])
event_schedule['start_date'] = pd.to_datetime(event_schedule['start_date'])
event_schedule['end_date'] = pd.to_datetime(event_schedule['end_date'])

# Nettoyage des données des streamers
streamers.rename(columns={'avg_viewers_zevent': 'avg_viewers_zevent', 'peak_viewers_zevent': 'peak_viewers_zevent'}, inplace=True)
streamers['donations_per_hour'].fillna(0, inplace=True)
streamers['donations_per_follower'].fillna(0, inplace=True)

# Remplacer les infinis par NaN, puis par 0
streamers.replace([np.inf, -np.inf], np.nan, inplace=True)
streamers.fillna(0, inplace=True)

# Afficher les informations sur les dataframes pour vérifier les types et les valeurs nulles
print("Informations sur les données des streamers :")
streamers.info()
print("\nInformations sur l'évolution des dons :")
donations_evolution.info()
print("\nInformations sur l'évolution des spectateurs :")
viewers_evolution.info()

Informations sur les données des streamers :
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 316 entries, 0 to 315
Data columns (total 26 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   twitch_login               316 non-null    object 
 1   name                       316 non-null    object 
 2   channel_name               316 non-null    object 
 3   total_donations            316 non-null    float64
 4   total_viewers_max          316 non-null    object 
 5   duration_seconds           316 non-null    float64
 6   donations_final            316 non-null    float64
 7   donations_start            316 non-null    float64
 8   avg_viewers_zevent         316 non-null    float64
 9   peak_viewers_zevent        316 non-null    float64
 10  watch_time_hours           316 non-null    float64
 11  stream_time_hours          316 non-null    float64
 12  peak_viewers_twitch        316 non-null    float64
 13  avg_v

C:\Users\Linwë\AppData\Local\Temp\ipykernel_19720\3696770416.py:9: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  streamers['donations_per_hour'].fillna(0, inplace=True)
C:\Users\Linwë\AppData\Local\Temp\ipykernel_19720\3696770416.py:10: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.


## 3. Analyse Exploratoire des Données (EDA)

### Statistiques Globales de l'Événement

In [4]:
# Statistiques globales
total_donations_event = donations_evolution['donations'].max()
peak_viewers_event = viewers_evolution['viewers'].max()
total_streamers = streamers['twitch_login'].nunique()

print(f"Total des dons récoltés : {total_donations_event:,.2f} €")
print(f"Pic de spectateurs simultanés : {peak_viewers_event:,.0f}")
print(f"Nombre total de streamers participants : {total_streamers}")

# Évolution des dons
fig = px.line(donations_evolution, x='datetime', y='donations', title="Évolution du Total des Dons au Fil du Temps",
              labels={'datetime': 'Date', 'donations': 'Dons (en millions €)'})
fig.show()

# Évolution des spectateurs
fig = px.line(viewers_evolution, x='datetime', y='viewers', title="Évolution du Nombre de Spectateurs au Fil du Temps",
              labels={'datetime': 'Date', 'viewers': 'Nombre de Spectateurs'})
fig.show()

Total des dons récoltés : 16,172,355.24 €
Pic de spectateurs simultanés : 752,185
Nombre total de streamers participants : 316


### Analyse des Streamers

In [5]:
# Top 10 des streamers par dons
top_10_donations = streamers.sort_values(by='total_donations', ascending=False).head(10)

fig = px.bar(top_10_donations, x='name', y='total_donations',
             title='Top 10 des Streamers par Total de Dons',
             labels={'name': 'Streamer', 'total_donations': 'Total des Dons (€)'},
             color='total_donations',
             color_continuous_scale=px.colors.sequential.Viridis)
fig.show()

# Top 10 des streamers par spectateurs moyens
top_10_avg_viewers = streamers.sort_values(by='avg_viewers_zevent', ascending=False).head(10)

fig = px.bar(top_10_avg_viewers, x='name', y='avg_viewers_zevent',
             title='Top 10 des Streamers par Spectateurs Moyens',
             labels={'name': 'Streamer', 'avg_viewers_zevent': 'Spectateurs Moyens'},
             color='avg_viewers_zevent',
             color_continuous_scale=px.colors.sequential.Plasma)
fig.show()

# Corrélation entre les spectateurs et les dons
fig = px.scatter(streamers, x='avg_viewers_zevent', y='total_donations',
                 title='Corrélation entre Spectateurs Moyens et Dons Totals',
                 labels={'avg_viewers_zevent': 'Spectateurs Moyens', 'total_donations': 'Total des Dons (€)'},
                 hover_name='name',
                 size='total_donations',
                 color='avg_viewers_zevent',
                 size_max=60)
fig.show()

### Analyse de l'Efficacité des Streamers

In [6]:
# Top 10 des streamers par dons par heure
top_10_donations_per_hour = streamers.sort_values(by='donations_per_hour', ascending=False).head(10)

fig = px.bar(top_10_donations_per_hour, x='name', y='donations_per_hour',
             title='Top 10 des Streamers par Dons par Heure',
             labels={'name': 'Streamer', 'donations_per_hour': 'Dons par Heure (€/h)'},
             color='donations_per_hour',
             color_continuous_scale=px.colors.sequential.Cividis_r)
fig.show()

# Top 10 des streamers par dons par follower
top_10_donations_per_follower = streamers.sort_values(by='donations_per_follower', ascending=False).head(10)

fig = px.bar(top_10_donations_per_follower, x='name', y='donations_per_follower',
             title='Top 10 des Streamers par Dons par Follower',
             labels={'name': 'Streamer', 'donations_per_follower': 'Dons par Follower (€)'},
             color='donations_per_follower',
             color_continuous_scale=px.colors.sequential.Inferno)
fig.show()

### Analyse des Objectifs de Dons

In [7]:
# Analyse des objectifs de dons
accomplished_goals = donation_goals['accomplished'].value_counts()

fig = px.pie(accomplished_goals, values=accomplished_goals.values, names=accomplished_goals.index,
             title='Proportion des Objectifs de Dons Atteints',
             labels={'names': 'Statut', 'values': 'Nombre d\'Objectifs'},
             color_discrete_map={True: 'green', False: 'red'})
fig.show()

# Relation entre les objectifs et les dons totaux
goals_per_streamer = donation_goals.groupby('streamer_name')['id'].count().reset_index()
goals_per_streamer.rename(columns={'id': 'goal_count'}, inplace=True)

streamers_with_goals = pd.merge(streamers, goals_per_streamer, left_on='name', right_on='streamer_name', how='left')
streamers_with_goals['goal_count'].fillna(0, inplace=True)

fig = px.scatter(streamers_with_goals, x='goal_count', y='total_donations',
                 title='Nombre d\'Objectifs par Streamer vs. Dons Totals',
                 labels={'goal_count': 'Nombre d\'Objectifs', 'total_donations': 'Total des Dons (€)'},
                 hover_name='name',
                 size='total_donations',
                 color='goal_count',
                 size_max=60)
fig.show()

C:\Users\Linwë\AppData\Local\Temp\ipykernel_19720\4246866410.py:15: FutureWarning:

A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.





## 6. Conclusion et Rapport d'Analyse

### Principales Conclusions

1.  **Performance Globale de l'Événement :**
    *   Le ZEvent 2025 a été un immense succès, récoltant un total de **16,17 millions d'euros**.
    *   L'événement a attiré un pic de **752 185 spectateurs simultanés**, démontrant son incroyable portée et son engagement communautaire.

2.  **Performance des Streamers :**
    *   **Leaders en Dons :** Des streamers comme **Domingo, AntoineDaniel, et ZeratoR** ont été les plus grands collecteurs de fonds, contribuant de manière significative au total.
    *   **Leaders en Audience :** Les mêmes noms dominent également les classements d'audience, indiquant une forte corrélation entre la popularité (mesurée par les spectateurs) et la capacité à générer des dons.
    *   **Corrélation Spectateurs-Dons :** L'analyse confirme une corrélation positive claire entre le nombre moyen de spectateurs et le total des dons. Les streamers avec une plus grande audience ont tendance à récolter plus de dons, bien qu'il existe des exceptions notables.

3.  **Efficacité des Streamers :**
    *   **Dons par Heure :** Certains streamers plus petits ont montré une efficacité remarquable en termes de dons par heure, suggérant un public très engagé et généreux.
    *   **Dons par Follower :** Cette métrique révèle des "champions cachés" qui, bien qu'ayant moins de followers, parviennent à mobiliser leur communauté de manière extrêmement efficace. Des streamers comme **AngleDroit** se démarquent ici.

4.  **Analyse Temporelle :**
    *   L'évolution des dons et des spectateurs montre des pics clairs pendant les événements clés du planning, en particulier lors des sessions de jeux populaires, des défis et des concerts.
    *   Les dernières heures de l'événement ("le rush final") voient une augmentation exponentielle des dons, une tendance classique mais toujours impressionnante.

5.  **Objectifs de Dons :**
    *   Une grande partie des objectifs de dons n'a pas été atteinte, ce qui pourrait indiquer qu'ils étaient très ambitieux ou que la dynamique des dons était concentrée sur d'autres aspects.
    *   Il existe une corrélation positive entre le nombre d'objectifs fixés par un streamer et le montant total des dons qu'il a récoltés, suggérant que les objectifs, même s'ils ne sont pas tous atteints, agissent comme un puissant moteur d'incitation.

### Recommandations et Perspectives

*   **Pour les Organisateurs :** Mettre en avant les "champions de l'efficacité" pourrait être une stratégie intéressante pour les futures éditions, afin de valoriser différentes formes de succès.
*   **Pour les Streamers :** Fixer de nombreux objectifs de dons, même ambitieux, semble être une stratégie payante pour stimuler les dons.
*   **Analyse Future :** Une analyse plus approfondie pourrait inclure une analyse de sentiment des discussions sur les réseaux sociaux, ou une modélisation prédictive pour estimer les dons potentiels basés sur les caractéristiques des streamers.

Ce notebook a fourni une analyse complète des données du ZEvent 2025, révélant des aperçus clés sur la performance de l'événement, l'impact des streamers, et les dynamiques d'engagement de la communauté.

# Analyse complète du ZEvent 2025
Ce notebook propose une analyse structurée : chargement, nettoyage, exploration et visualisations interactives.

## 1. Importation et configuration
```python
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
from datetime import datetime

# Styles et options
sns.set_theme(style='whitegrid')
pd.options.display.max_columns = None
pd.options.display.max_rows = 100
```

In [ ]:
# 2. Chargement et aperçu des fichiers de données
import glob
file_list = glob.glob('../data/clean/**/**/*.csv', recursive=True)

# Lire les 10 premières lignes de chaque fichier
preview = {}
for f in file_list:
    preview[f] = pd.read_csv(f, nrows=10)
# Afficher un dataframe combiné de noms de fichiers et dimensions
pd.DataFrame({'file': list(preview.keys()), 'rows': [df.shape[0] for df in preview.values()], 'cols': [df.shape[1] for df in preview.values()]})

## 3. Nettoyage et préparation des données
```python
# Convertir datetime dans les objets preview
for name, df in preview.items():
    for col in df.columns:
        if 'date' in col.lower() or 'time' in col.lower():
            df[col] = pd.to_datetime(df[col], errors='coerce')
```

In [ ]:
# Charger l'intégralité des jeux de données pour analyse
donation_goals = pd.read_csv('../data/clean/events/donation_goals.csv')
event_schedule = pd.read_csv('../data/clean/events/event_schedule.csv')
streamers_data = pd.read_csv('../data/clean/streamers/streamers_data.csv')
streamers_detailed_stats = pd.read_csv('../data/clean/streamers/streamers_detailed_stats.csv')
donations_evolution = pd.read_csv('../data/clean/temporal/donations_evolution.csv')
viewers_evolution = pd.read_csv('../data/clean/temporal/viewers_evolution.csv')

# Conversion des dates dans les DataFrames principaux
for df in [donation_goals, event_schedule, donations_evolution, viewers_evolution]:
    for col in df.columns:
        if 'date' in col.lower() or 'time' in col.lower() or 'datetime' in col.lower():
            df[col] = pd.to_datetime(df[col], errors='coerce')

## 5. Analyse temporelle des dons et spectateurs
```python
# Tracé des évolutions temporelles
plt.figure(figsize=(12,6))
sns.lineplot(data=donations_evolution, x='datetime', y='donations', label='Dons')
sns.lineplot(data=viewers_evolution, x='datetime', y='viewers', label='Viewers')
plt.title('Évolution des dons et viewers au fil du temps')
plt.xlabel('Date')
plt.ylabel('Valeur')
plt.legend()
plt.show()
```

## 6. Comparaison des performances des streamers
```python
top_streamers = streamers_detailed_stats.nlargest(10, 'total_donations')
fig = px.bar(top_streamers, x='name', y='total_donations', title='Top 10 des streamers par dons totaux', color='total_donations', color_continuous_scale='Viridis')
fig.show()
```

## 7. Corrélation planning vs activité
```python
fig, ax1 = plt.subplots(figsize=(14,6))
ax1.plot(donations_evolution['datetime'], donations_evolution['donations'], color='tab:blue', label='Dons')
ax1.set_ylabel('Dons', color='tab:blue')
ax2 = ax1.twinx()
ax2.plot(viewers_evolution['datetime'], viewers_evolution['viewers'], color='tab:green', label='Viewers')
ax3 = ax1.twinx()
ax3.spines['right'].set_position(('axes', 1.1))
ax3.plot(event_schedule['start'], event_schedule['event'], color='tab:red', linestyle='--', label='Planning')
ax3.set_ylabel('Événement', color='tab:red')
ax1.set_xlabel('Date')
ax1.set_title('Corrélation entre planning et activité')
fig.legend()
plt.show()
```

## 8. Visualisations interactives avec Plotly Express
Explorez dynamiquement les données avec des graphiques interactifs.

In [ ]:
# Graphique interactif: évolution cumulative des dons
fig = px.area(donations_evolution, x='datetime', y='donations', title='Dons évolutifs', labels={'donations':'Montant (€)','datetime':'Date'})
fig.update_layout(hovermode='x')
fig.show()

# Scatter interactif: donations vs viewers
fig = px.scatter(donations_evolution.merge(viewers_evolution, on='datetime'), x='viewers', y='donations', color='donations', size='viewers', title='Dons vs Viewers')
fig.show()

# Bar interactif: top streamers
fig = px.bar(top_streamers, x='name', y='total_donations', title='Top Streamers (Interactif)', labels={'total_donations':'Dons totaux (€)'}, hover_data=['avg_viewers'])
fig.show()

## 4. Analyse descriptive des données
```python
# Statistiques de base sur les dons et viewers
print("Statistiques des dons :")
print(donations_evolution['donations'].describe())
print("\nStatistiques des viewers :")
print(viewers_evolution['viewers'].describe())

# Statistiques par streamer
print("\nTop 10 des streamers par donations totales :")
print(streamers_detailed_stats[['name','total_donations','avg_viewers']].sort_values('total_donations', ascending=False).head(10))
```